src/decycle.ts

```diff
@@ -28,6 +28,7 @@ export const decycle = (obj: unknown, replacer?: (value: any) => any) => {
           curObj[name] = derez(value[name], `${path}/${name}`);
         });
       }
+      objs.delete(value);
```

Comment:
Hmm, this looked too simple to be true.

```ts
  it('should decyyy', () => {
    const obj = {
      names: {
        get foo() {
          return this.bar;
        },
        get bar() {
          return obj.cave;
        },
      },
      cave: {},
      get bears() {
        return this.cave;
      },
    };

    expect(decycle(obj)).toEqual({
      bears: {
        $ref: '#/names/foo',
      },
      cave: {
        $ref: '#/names/foo',
      },
      names: {
        bar: {
          $ref: '#/names/foo',
        },
        foo: {},
      },
    });
  });
```  

![image](https://user-images.githubusercontent.com/9273484/117352508-5bf19180-aeaf-11eb-8d37-4e68b15b165a.png)

Reply:
the left is before the change (so current master) and the after is your PR